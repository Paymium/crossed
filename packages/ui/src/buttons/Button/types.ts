/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { CrossedMethods } from '@crossed/styled';
import { PressableProps } from 'react-native';
import { alignSelfStyle } from '../../styles/alignItems';
import { buttonSizeStyles } from './styles';

type Variants =
  | 'primary'
  | 'tertiary'
  | 'secondary'
  | 'success'
  | 'error'
  | false;
export interface ButtonProps extends Omit<PressableProps, 'style'> {
  /**
   * Select button variant
   * @default primary
   */
  variant?: Variants;

  /**
   * Loading state of button
   * @type {boolean}
   */
  loading?: boolean;

  /**
   * Extend style of button
   * @type {CrossedMethods<any>}
   */
  // style?: CrossedMethods<any>;

  /**
   * Disable size
   * @default md
   */
  size?: Exclude<keyof typeof buttonSizeStyles, 'default'>;

  /**
   * Set align-self style
   */
  alignSelf?: keyof typeof alignSelfStyle;

  /**
   * Error style with variant
   */
  error?: boolean;

  /**
   * Success style with variant
   */
  success?: boolean;

  style?: CrossedMethods<any>;
}
