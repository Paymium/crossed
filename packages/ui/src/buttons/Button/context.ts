/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createContext } from 'react';
import { ButtonProps } from './types';

export const buttonContext = createContext<
  Pick<ButtonProps, 'variant' | 'size' | 'disabled' | 'error'> & {
    state?: {
      active?: boolean;
      hover?: boolean;
    };
    textId: string;
    // eslint-disable-next-line no-unused-vars
    setTextId: (p: string) => void;
  }
>({} as any);
