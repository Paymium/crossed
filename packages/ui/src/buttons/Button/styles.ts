/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createStyles, type StyleSheet } from '@crossed/styled';

type RequiredKeys = {
  root: StyleSheet;
  text: StyleSheet;
  textHover: StyleSheet;
  textActive: StyleSheet;
};

export const buttonSizeStyles = createStyles(({ space }) => ({
  default: {
    ':disabled': { opacity: 0.5 },
    web: {
      'base': { boxSizing: 'border-box' },
      ':focus-visible': {
        outlineWidth: '2px',
        outlineOffset: '2px',
        outlineStyle: 'solid',
      },
    },
    base: {
      display: 'flex',
      borderRadius: 7,
      borderWidth: 2,
      borderStyle: 'solid',
      borderColor: 'transparent',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: 'transparent',
      gap: 8,
    },
  },
  sm: { base: { height: 32, paddingHorizontal: space.xs } },
  md: { base: { height: 44, paddingHorizontal: space.md } },
  lg: { base: { height: 52, paddingHorizontal: space.lg } },
}));
export const buttonPrimaryStyles = createStyles(
  ({ components: { Action } }) =>
    ({
      root: {
        'base': {
          backgroundColor: Action.primary.default.background,
          borderColor: Action.primary.default.background,
        },
        ':hover': {
          backgroundColor: Action.primary.hover.background,
          borderColor: Action.primary.hover.background,
        },
        ':active': {
          backgroundColor: Action.primary.active.background,
          borderColor: Action.primary.active.background,
        },
        'web': {
          ':focus-visible': {
            outlineColor: Action.primary.focus.border,
          },
        },
      },
      text: { 'base': { color: Action.primary.default.text } },
      textHover: { 'base': { color: Action.primary.hover.text } },
      textActive: { 'base': { color: Action.primary.active.text } },
    }) satisfies RequiredKeys
);

export const buttonSecondaryStyles = createStyles(
  ({ components: { Action } }) =>
    ({
      root: {
        'base': {
          borderColor: Action.secondary.default.border,
          backgroundColor: Action.secondary.default.background,
        },
        ':hover': {
          borderColor: Action.secondary.hover.border,
          backgroundColor: Action.secondary.hover.background,
        },
        ':active': {
          borderColor: Action.secondary.active.border,
          backgroundColor: Action.secondary.active.background,
        },
        'web': {
          ':focus-visible': {
            outlineColor: Action.secondary.focus.border,
          },
        },
      },
      text: { 'base': { color: Action.secondary.default.text } },
      textHover: { 'base': { color: Action.secondary.hover.text } },
      textActive: { 'base': { color: Action.secondary.active.text } },
    }) satisfies RequiredKeys
);

export const buttonTertiaryStyles = createStyles(
  ({ components: { Action } }) =>
    ({
      root: {
        'base': {
          borderColor: Action.tertiary.default.border,
          backgroundColor: Action.tertiary.default.background,
        },
        ':hover': {
          borderColor: Action.tertiary.hover.border,
          backgroundColor: Action.tertiary.hover.background,
        },
        ':active': {
          borderColor: Action.tertiary.active.border,
          backgroundColor: Action.tertiary.active.background,
        },
        'web': {
          ':focus-visible': {
            outlineColor: Action.tertiary.focus.border,
            backgroundColor: Action.tertiary.focus.background,
          },
        },
      },
      text: { 'base': { color: Action.tertiary.default.text } },
      textHover: { 'base': { color: Action.tertiary.hover.text } },
      textActive: { 'base': { color: Action.tertiary.active.text } },
    }) satisfies RequiredKeys
);

export const buttonSuccessStyles = createStyles(
  ({ colors }) =>
    ({
      root: {
        'base': {
          borderColor: colors.success.primary,
          backgroundColor: colors.success.primary,
        },
        ':hover': {
          borderColor: colors.success.dark,
          backgroundColor: colors.success.dark,
        },
        ':active': {
          borderColor: colors.success.dark,
          backgroundColor: colors.success.dark,
        },
        'web': {
          ':focus-visible': {
            outlineColor: colors.success.primary,
          },
        },
      },
      text: { 'base': { color: colors.white } },
      textHover: {},
      textActive: {},
    }) satisfies RequiredKeys
);

export const buttonErrorStyles = createStyles(
  ({ colors }) =>
    ({
      root: {
        'base': {
          backgroundColor: colors.error.primary,
          borderColor: colors.error.primary,
        },
        ':hover': {
          backgroundColor: colors.error.muted,
          borderColor: colors.error.muted,
        },
        ':active': {
          backgroundColor: colors.error.satured,
          borderColor: colors.error.satured,
        },
        'web': {
          ':focus-visible': {
            outlineColor: colors.error.primary,
          },
        },
      },
      text: { 'base': { color: colors.white } },
      textHover: {},
      textActive: {},
    }) satisfies RequiredKeys
);

export const textStyles = createStyles(() => ({
  default: { base: { fontWeight: 'bold' } },
  disabled: { base: { pointerEvents: 'none' } },
  // sm: { base: { height: 32, paddingHorizontal: space.xs } },
  // md: { base: { height: 44, paddingHorizontal: space.md } },
  // lg: { base: { height: 52, paddingHorizontal: space.lg } },
}));
