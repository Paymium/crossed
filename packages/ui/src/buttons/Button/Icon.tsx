/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { composeStyles, CrossedMethods } from '@crossed/styled';
import {
  cloneElement,
  isValidElement,
  PropsWithChildren,
  useContext,
} from 'react';
import { buttonContext } from './context';
import {
  buttonPrimaryErrorStyle,
  buttonPrimaryStyles,
  buttonSecondaryErrorStyle,
  buttonSecondaryStyles,
  buttonTertiaryErrorStyle,
  buttonTertiaryStyles,
  textStyles,
} from './styles';

export const ButtonIcon = ({
  children,
  style,
}: PropsWithChildren<{ style?: CrossedMethods<any> }>) => {
  const { variant, state, disabled, error } = useContext(buttonContext);
  const { hover, active } = state;

  const color = composeStyles(
    textStyles.default,
    variant === 'primary' &&
      composeStyles(
        ...(!error
          ? [
              buttonPrimaryStyles.text,
              hover && buttonPrimaryStyles.textHover,
              active && buttonPrimaryStyles.textActive,
            ]
          : [
              buttonPrimaryErrorStyle.text,
              hover && buttonPrimaryErrorStyle.textHover,
              active && buttonPrimaryErrorStyle.textActive,
            ])
      ),
    variant === 'secondary' &&
      composeStyles(
        ...(!error
          ? [
              buttonSecondaryStyles.text,
              hover && buttonSecondaryStyles.textHover,
              active && buttonSecondaryStyles.text,
            ]
          : [
              buttonSecondaryErrorStyle.text,
              hover && buttonSecondaryErrorStyle.textHover,
              active && buttonSecondaryErrorStyle.textActive,
            ])
      ),
    variant === 'tertiary' &&
      composeStyles(
        ...(!error
          ? [
              buttonTertiaryStyles.text,
              hover && buttonTertiaryStyles.textHover,
              active && buttonTertiaryStyles.textActive,
            ]
          : [
              buttonTertiaryErrorStyle.text,
              hover && buttonTertiaryErrorStyle.textHover,
              active && buttonTertiaryErrorStyle.textActive,
            ])
      ),
    disabled && textStyles.disabled,
    style
  ).style(state).style.color;

  return isValidElement(children)
    ? cloneElement(children, { color } as any)
    : children;
};
ButtonIcon.displayName = 'Button.Icon';
