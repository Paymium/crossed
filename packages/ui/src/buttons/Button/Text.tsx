/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { forwardRef, useContext, useEffect } from 'react';
import { Text, type TextProps } from '../../typography/Text';
import { buttonContext } from './context';
import { composeStyles } from '@crossed/styled';
import {
  buttonErrorStyles,
  buttonPrimaryStyles,
  buttonSecondaryStyles,
  buttonSuccessStyles,
  buttonTertiaryStyles,
  textStyles,
} from './styles';
import { Text as TextNative } from 'react-native';

type ButtonTextProps = TextProps;

const ButtonText = forwardRef<TextNative, ButtonTextProps>((props, ref) => {
  const { variant, state, disabled, setTextId, textId } =
    useContext(buttonContext);

  const { hover, active } = state || {};

  useEffect(() => {
    if (props.id && textId !== props.id) {
      setTextId(props.id);
    }
  }, [props.id, textId, setTextId]);

  return (
    <Text
      fontWeight={'lg'}
      {...props}
      id={textId}
      style={composeStyles(
        textStyles.default,
        variant === 'primary' && buttonPrimaryStyles.text,
        variant === 'primary' && hover && buttonPrimaryStyles.textHover,
        variant === 'primary' && active && buttonPrimaryStyles.textActive,
        variant === 'secondary' && buttonSecondaryStyles.text,
        variant === 'secondary' && hover && buttonSecondaryStyles.textHover,
        variant === 'secondary' && active && buttonSecondaryStyles.text,
        variant === 'tertiary' && buttonTertiaryStyles.text,
        variant === 'tertiary' && hover && buttonTertiaryStyles.textHover,
        variant === 'tertiary' && active && buttonTertiaryStyles.textActive,
        variant === 'error' && buttonErrorStyles.text,
        variant === 'success' && buttonSuccessStyles.text,
        disabled && textStyles.disabled,
        props.style
      )}
      ref={ref}
    />
  );
});
ButtonText.displayName = 'Button.Text';

export { ButtonText, type ButtonTextProps };
