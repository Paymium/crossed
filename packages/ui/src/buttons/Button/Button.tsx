/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

'use client';

import { composeStyles } from '@crossed/styled';
import { ActivityIndicator, Pressable, View } from 'react-native';
import { forwardRef, useId, useState } from 'react';
import {
  buttonOutlineStyle,
  buttonPrimaryErrorStyle,
  buttonPrimaryStyles,
  buttonPrimarySuccessStyle,
  buttonSecondaryErrorStyle,
  buttonSecondaryStyles,
  buttonSizeStyles,
  buttonStyle,
  buttonTertiaryErrorStyle,
  buttonTertiaryStyles,
} from './styles';
import { ButtonProps } from './types';
import { ButtonIcon } from './Icon';
import { buttonContext } from './context';
import { alignSelfStyle } from '../../styles/alignItems';

export const Button = forwardRef<View, ButtonProps>(
  (
    {
      variant = 'primary',
      size = 'md',
      error,
      success,
      disabled,
      loading,
      children,
      alignSelf,
      ...props
    },
    ref
  ) => {
    const renderLoading = loading ? (
      <ButtonIcon>
        <ActivityIndicator />
      </ButtonIcon>
    ) : null;
    const id = useId();
    const [textId, setTextId] = useState(id);

    return (
      <Pressable
        aria-disabled={Boolean(disabled ?? false)}
        aria-labelledby={textId}
        role="button"
        tabIndex={disabled ? -1 : 0}
        disabled={disabled || loading}
        ref={ref}
        {...props}
        style={(e: any) =>
          composeStyles(
            buttonStyle,
            buttonOutlineStyle.default,
            size && buttonSizeStyles[size],
            alignSelf && alignSelfStyle[alignSelf],
            variant === 'primary' &&
              composeStyles(
                ...(error
                  ? [
                      buttonPrimaryErrorStyle.root,
                      buttonOutlineStyle.error,
                      disabled && buttonPrimaryErrorStyle.disabled,
                    ]
                  : success
                    ? [
                        buttonPrimarySuccessStyle.root,
                        buttonOutlineStyle.success,
                        disabled && buttonPrimarySuccessStyle.disabled,
                      ]
                    : [
                        buttonPrimaryStyles.root,
                        buttonOutlineStyle.primary,
                        disabled && buttonPrimaryStyles.disabled,
                      ])
              ),
            variant === 'secondary' &&
              composeStyles(
                ...(error
                  ? [
                      buttonSecondaryErrorStyle.root,
                      buttonOutlineStyle.error,
                      disabled && buttonSecondaryErrorStyle.disabled,
                    ]
                  : [
                      buttonSecondaryStyles.root,
                      buttonOutlineStyle.primary,
                      disabled && buttonSecondaryStyles.disabled,
                    ])
              ),
            variant === 'tertiary' &&
              composeStyles(
                ...(error
                  ? [buttonTertiaryErrorStyle.root, buttonOutlineStyle.error]
                  : [
                      buttonTertiaryStyles.root,
                      buttonOutlineStyle.primary,
                      disabled && buttonTertiaryStyles.disabled,
                    ])
              ),
            props.style
          ).rnw({
            hover: (!disabled && e.hovered) || loading,
            active: !disabled && !loading && e.pressed,
            disabled: disabled,
          }).style
        }
      >
        {(e: any) => {
          return (
            <>
              <buttonContext.Provider
                value={{
                  variant,
                  size,
                  state: {
                    active: e.pressed,
                    hover: e.hovered || loading,
                  },
                  disabled: disabled,
                  textId,
                  setTextId,
                  error,
                  success,
                }}
              >
                {renderLoading}
                {typeof children === 'function' ? children(e) : children}
              </buttonContext.Provider>
            </>
          );
        }}
      </Pressable>
    );
  }
);
Button.displayName = 'Button';
