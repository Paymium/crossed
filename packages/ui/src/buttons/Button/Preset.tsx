/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { Button } from './Button';
import { ButtonText } from './Text';
import { ComponentProps } from 'react';
import { ButtonIcon } from './Icon';

export type ButtonPresetProps = ComponentProps<typeof Button> & {
  /**
   * Text of button
   */
  text?: string;
  /**
   * Icon right
   */
  iconRight?: React.ReactNode;
  /**
   * Icon left
   */
  iconLeft?: React.ReactNode;
};

export const ButtonPreset = ({
  text,
  iconRight,
  iconLeft,
  ...props
}: ButtonPresetProps) => {
  return (
    <Button {...props}>
      {!!iconLeft && <ButtonIcon>{iconLeft}</ButtonIcon>}
      {text && <ButtonText>{text}</ButtonText>}
      {!!iconRight && <ButtonIcon>{iconRight}</ButtonIcon>}
    </Button>
  );
};

ButtonPreset.displayName = 'ButtonPreset';
