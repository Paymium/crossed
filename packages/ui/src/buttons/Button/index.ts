/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { withStaticProperties } from '@crossed/core';
import { ButtonText, ButtonTextProps } from './Text';
import { ButtonGroup, ButtonGroupProps } from './Group';
import { Box as ButtonElement } from '../../layout/Box';
import { ButtonIcon } from './Icon';
import { Button as Root } from './Button';

export * from './styles';
export * from './types';

export const Button = withStaticProperties(Root, {
  Group: ButtonGroup,
  Text: ButtonText,
  Element: ButtonElement,
  Icon: ButtonIcon,
});
export { ButtonText, ButtonElement, ButtonGroup, ButtonIcon };
export type { ButtonTextProps, ButtonGroupProps };
