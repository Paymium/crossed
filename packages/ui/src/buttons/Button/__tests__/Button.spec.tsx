/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import React from 'react';
import { render, screen, fireEvent } from '@crossed/test';
import { Button } from '../Button';

describe('Button', () => {
  it('renders correctly with default props', () => {
    render(<Button>Click Me</Button>);
    const button = screen.getByRole('button');
    expect(button).toBeTruthy();
    expect(button).not.toHaveAttribute('aria-disabled');
  });

  it('renders children correctly', () => {
    render(<Button>Click Me</Button>);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('Click Me');
  });

  it('disables the button when `disabled` is true', () => {
    render(<Button disabled>Click Me</Button>);
    const button = screen.getByRole('button');
    expect(button).toHaveAttribute('aria-disabled', 'true');
    expect(button).toBeDisabled();
  });

  it('renders the loading indicator when `loading` is true', () => {
    render(<Button loading>Click Me</Button>);
    const loadingIndicator = screen.getByRole('progressbar'); // ActivityIndicator has `progressbar` role
    expect(loadingIndicator).toBeTruthy();

    expect(screen.getByRole('button')).toHaveTextContent('Click Me');
  });

  // it('renders children as a function and provides event state', () => {
  //   const mockChild = jest.fn(({ hovered, pressed }: any) => (
  //     <span>{hovered ? 'Hovered' : pressed ? 'Pressed' : 'Idle'}</span>
  //   ));
  //
  //   render(<Button>{mockChild}</Button>);
  //   const button = screen.getByRole('button');
  //
  //   fireEvent.mouseOver(button);
  //   expect(mockChild).toHaveBeenCalledWith(
  //     expect.objectContaining({
  //       'focused': false,
  //       hovered: true,
  //       pressed: false,
  //     })
  //   );
  //
  //   fireEvent.mouseDown(button);
  //   expect(mockChild).toHaveBeenCalledWith(
  //     expect.objectContaining({ hovered: true, pressed: true })
  //   );
  //
  //   fireEvent.mouseLeave(button);
  //   expect(mockChild).toHaveBeenCalledWith(
  //     expect.objectContaining({ hovered: false, pressed: false })
  //   );
  // });

  it('applies the correct styles for primary variant', () => {
    render(<Button variant="primary">Primary</Button>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('applies the correct styles for secondary variant', () => {
    render(<Button variant="secondary">Secondary</Button>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('applies the correct styles for lg size', () => {
    render(<Button size="lg">Large Button</Button>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('applies the correct styles for sm size', () => {
    render(<Button size="sm">Small Button</Button>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('supports forwarding refs', () => {
    const ref = React.createRef<HTMLButtonElement>();
    render(<Button ref={ref as any}>Click Me</Button>);
    expect(ref.current).toBeTruthy();
  });

  it('handles dynamic states (hover and active)', () => {
    render(<Button>Dynamic Button</Button>);
    const button = screen.getByRole('button');

    fireEvent.mouseOver(button);
    expect(button).not.toHaveAttribute('aria-disabled');

    fireEvent.mouseDown(button);
    expect(button).not.toHaveAttribute('aria-disabled');
  });
});
