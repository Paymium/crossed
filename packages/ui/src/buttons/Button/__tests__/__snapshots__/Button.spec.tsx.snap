// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button applies the correct styles for lg size 1`] = `
<button
  aria-labelledby=":r6:"
  class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73 disabled:opacity-[0.5] box-sizing-[border-box] focus-visible:outline-width-[2px] focus-visible:outline-offset-[2px] focus-visible:outline-style-[solid] focus-visible:outline-color-[var(--components--action-primary-focus-border)] display-[flex] border-radius-[7px] border-width-[2px] border-style-[solid] border-color-[var(--components--action-primary-default-background)] flex-direction-[row] align-items-[center] justify-content-[center] background-color-[var(--components--action-primary-default-background)] gap-[8px] height-[52px] padding-horizontal-[var(--space-lg)] hover:background-color-[var(--components--action-primary-hover-background)] hover:border-color-[var(--components--action-primary-hover-background)] active:background-color-[var(--components--action-primary-active-background)] active:border-color-[var(--components--action-primary-active-background)]"
  role="button"
  tabindex="0"
  type="button"
>
  Large Button
</button>
`;

exports[`Button applies the correct styles for primary variant 1`] = `
<button
  aria-labelledby=":r4:"
  class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73 disabled:opacity-[0.5] box-sizing-[border-box] focus-visible:outline-width-[2px] focus-visible:outline-offset-[2px] focus-visible:outline-style-[solid] focus-visible:outline-color-[var(--components--action-primary-focus-border)] display-[flex] border-radius-[7px] border-width-[2px] border-style-[solid] border-color-[var(--components--action-primary-default-background)] flex-direction-[row] align-items-[center] justify-content-[center] background-color-[var(--components--action-primary-default-background)] gap-[8px] height-[44px] padding-horizontal-[var(--space-md)] hover:background-color-[var(--components--action-primary-hover-background)] hover:border-color-[var(--components--action-primary-hover-background)] active:background-color-[var(--components--action-primary-active-background)] active:border-color-[var(--components--action-primary-active-background)]"
  role="button"
  tabindex="0"
  type="button"
>
  Primary
</button>
`;

exports[`Button applies the correct styles for secondary variant 1`] = `
<button
  aria-labelledby=":r5:"
  class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73 disabled:opacity-[0.5] box-sizing-[border-box] focus-visible:outline-width-[2px] focus-visible:outline-offset-[2px] focus-visible:outline-style-[solid] focus-visible:outline-color-[var(--components--action-secondary-focus-border)] display-[flex] border-radius-[7px] border-width-[2px] border-style-[solid] border-color-[var(--components--action-secondary-default-border)] flex-direction-[row] align-items-[center] justify-content-[center] background-color-[var(--components--action-secondary-default-background)] gap-[8px] height-[44px] padding-horizontal-[var(--space-md)] hover:border-color-[var(--components--action-secondary-hover-border)] hover:background-color-[var(--components--action-secondary-hover-background)] active:border-color-[var(--components--action-secondary-active-border)] active:background-color-[var(--components--action-secondary-active-background)]"
  role="button"
  tabindex="0"
  type="button"
>
  Secondary
</button>
`;

exports[`Button applies the correct styles for sm size 1`] = `
<button
  aria-labelledby=":r7:"
  class="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73 disabled:opacity-[0.5] box-sizing-[border-box] focus-visible:outline-width-[2px] focus-visible:outline-offset-[2px] focus-visible:outline-style-[solid] focus-visible:outline-color-[var(--components--action-primary-focus-border)] display-[flex] border-radius-[7px] border-width-[2px] border-style-[solid] border-color-[var(--components--action-primary-default-background)] flex-direction-[row] align-items-[center] justify-content-[center] background-color-[var(--components--action-primary-default-background)] gap-[8px] height-[32px] padding-horizontal-[var(--space-xs)] hover:background-color-[var(--components--action-primary-hover-background)] hover:border-color-[var(--components--action-primary-hover-background)] active:background-color-[var(--components--action-primary-active-background)] active:border-color-[var(--components--action-primary-active-background)]"
  role="button"
  tabindex="0"
  type="button"
>
  Small Button
</button>
`;
